src/shared/backend/services/cloudbuild.ts

```diff
@@ -88,6 +82,46 @@ export const createJob = async (params: JobStatusParams): Promise<Job> => {
   return data as Job;
 };
 
+function timeout(ms: number): Promise<void> {
+  /* eslint-disable-next-line no-promise-executor-return */
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+export const waitForJobSuccess = async (
+  params: JobStatusParams,
+  updateStatus: (statusData: {
+    jobStatus: JobStatus;
+    startedAt: string;
+  }) => void,
+  timeoutMs = 960000 // 16mn
+): Promise<Job | undefined> => {
+  const iterTime = 5000;
+  const iterNb = Math.ceil(timeoutMs / iterTime);
+
+  // job status data
+  let startedAt = new Date().getTime().toString();
+  let lastStatus: JobStatus | undefined;
+
+  /* eslint-disable no-await-in-loop */
+  for (let i = 0; i < iterNb; i += 1) {
```

Comment:
You don't need to do it this way btw, as it doesn't take into account slow network.

Instead:
```typescript
let timedOut = false;
const interval = setTimeout(() => {
  timedOut = true;
  reject(new Error("Build process timeout"));
}, timeout);
return new Promise((resolve, reject) => {
  (async () => {
    while (!timedOut) {
      const jobStatus = await queryJobStatus(params);

      // notify client of current status
      if (lastStatus !== jobStatus.status && !timedOut) {
        startedAt = new Date().getTime().toString();
        lastStatus = jobStatus.status;
      }
      if (!timedOut) {
        updateStatus({ jobStatus: jobStatus.status, startedAt });
      }

      if (jobStatus.status === "BUILD_SUCCESS" && !timedOut) {
        resolve(jobStatus);
        return;
      }
      await timeout(iterTime);  
    }
  })().then(resolve).catch(reject).finally(() => clearTimeout(interval));
});
```

Reply:
would this be okay?

```typescript
  const controller = new AbortController();
  const timeoutId = setTimeout(() => {
    controller.abort();
    timedOut = true;
  }, timeoutMs);

  /* eslint-disable no-await-in-loop */
  while (!timedOut) {
    try {
      const jobStatus = await queryJobStatus(params, controller.signal);

      // notify client of current status
      if (lastStatus !== jobStatus.status) {
        startedAt = new Date().getTime().toString();
        lastStatus = jobStatus.status;
      }
      updateStatus({ jobStatus: jobStatus.status, startedAt });

      if (jobStatus.status === "BUILD_SUCCESS") {
        clearTimeout(timeoutId);
        return jobStatus;
      }

      await timeout(iterTime);
    } catch (err) {
      // ignore controller abort error, it's timeout
      if (err instanceof Error && err.name === "AbortError") {
        console.error(err);
      } else {
        throw err;
      }
    }
  }
  /* eslint-enable no-await-in-loop */
  clearTimeout(timeoutId);
  throw new Error("Build process timeout");
```